app/views/exercises/_subnav.html.erb

```diff
@@ -7,7 +7,7 @@
   <%= subnav_item(:widget, "Services", term_exercise_services_path(exercise.term, exercise), params[:controller] == "services")  if policy(ServicePolicy.term_policy_record(exercise.term)).index? %>
 
   <%= subnav_item(:list, "Submission", exercise_student_submission_path(exercise), params[:controller] == "submissions" || params[:controller] == "submission_tree") if current_account.student_of_term?(exercise.term) %>
-  <%= subnav_item(:megaphone, "Results", term_result_path(exercise.term, exercise), params[:controller] == "student_results") if policy(StudentResultsPolicy.term_policy_record(exercise.term)).index? %>
+  <%= subnav_item(:megaphone, "Results", term_result_path(exercise.term, exercise), params[:controller] == "student_results") if policy(StudentResultsPolicy.term_policy_record(exercise.term)).index? && GradingReview::SubmissionReview.find_by_account_and_exercise(current_account, exercise).published? %>
```

Comment:
Have you tested this? I believe there will be an error raised when the student hasn't created a submission for this exercise. You might want to do this check differently. For example:

```ruby
Submission.active.for_account(current_account).for_exercise(exercise).first.try(:result_published?)
```

I would also like to see this functionality wrapped in a method, like `exercise.results_published_for_account?(current_account)`, but I'm open to your suggestions and API.